cmd/snap-device-helper/snap-device-helper-test.c

```diff
@@ -282,6 +282,94 @@ static void test_sdh_action_nvme(sdh_test_fixture *fixture, gconstpointer test_d
+    /* check that fallback guessing of device type if applied during remove action */
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1p1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/ng0n1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/hwmon0");
+    mkdir_in_sysroot(fixture, "/sys/devices/foo/block/sda/sda4");
+    mkdir_in_sysroot(fixture, "/sys//devices/pnp0/00:04/tty/ttyS0");
+
+    struct {
+        const char *dev;
+        const char *majmin;
+        int expected_maj;
+        int expected_min;
+        int expected_type;
+    } tcs[] = {
+        /* these device paths match the fallback pattern of block devices */
+        {
+            .dev = "/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1",
+            .majmin = "259:0",
+            .expected_maj = 259,
+            .expected_min = 0,
+            .expected_type = S_IFBLK,
+        },
+        {
+            .dev = "/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1p1",
+            .majmin = "259:1",
+            .expected_maj = 259,
+            .expected_min = 1,
+            .expected_type = S_IFBLK,
+        },
+        {
+            .dev = "/devices/foo/block/sda/sda4",
+            .majmin = "8:0",
+            .expected_maj = 8,
+            .expected_min = 0,
+            .expected_type = S_IFBLK,
+        },
+        /* these are treated as char devices */
+        {
+            .dev = "/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0",
+            .majmin = "242:0",
+            .expected_maj = 242,
+            .expected_min = 0,
+            .expected_type = S_IFCHR,
+        },
+        {
+            .dev = "/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/hwmon0",
+            .majmin = "241:0",
+            .expected_maj = 241,
+            .expected_min = 0,
+            .expected_type = S_IFCHR,
+        },
+        {
+            .dev = "/devices/pnp0/00:04/tty/ttyS0",
+            .majmin = "4:64",
+            .expected_maj = 4,
+            .expected_min = 64,
+            .expected_type = S_IFCHR,
+        },
+    };
+
+    int bogus = 0;
+
+    for (size_t i = 0; i < sizeof(tcs) / sizeof(tcs[0]); i++) {
+        mocks_reset();
+        /* make cgroup_device_new return a non-NULL */
+        mocks.new_ret = &bogus;
+
+        struct sdh_invocation inv_block = {
+            .action = "remove",
+            .tagname = "snap_foo_bar",
+            .devpath = tcs[i].dev,
+            .majmin = tcs[i].majmin,
+        };
+        int ret = snap_device_helper_run(&inv_block);
+        g_assert_cmpint(ret, ==, 0);
+        g_assert_cmpint(mocks.cgorup_new_calls, ==, 1);
```

Comment:
heh actually that is the name of that variable already - so the original code has this typo too.

Reply:
It might be worth a look at https://github.com/check-spelling/check-spelling at some point :) a friend has found some pretty surprising bugs. This one's a bit harmless, I think.

Reply:
This is fixed in #11193. Thanks for he pointer, the check-spelling action looks nice, I'll look into integrating this with the snapd github workflow.